corehq/apps/domain/tests/test_auth_decorators.py

```diff
@@ -63,6 +64,19 @@ def _get_request(self, user):
         request.user = user
         return request
 
+    def test_no_user_set(self):
+        # no matter what, no user = no success
+        for allow_cc_users in (True, False):
+            for allow_sessions in (True, False):
+                for require_domain in (True, False):
+                    decorator = _login_or_challenge(passing_decorator,
+                                                    allow_cc_users=allow_cc_users,
+                                                    allow_sessions=allow_sessions,
+                                                    require_domain=require_domain)
+                    test = decorator(sample_view)
+                    request = self._get_request(user=AnonymousUser())
+                    self.assertNotEqual(SUCCESS, test(request, self.domain_name))
```

Comment:
What do you think of moving this check out to a new assertion function? Something like
```py
def assertForbidden(self, view_result):
    self.assertNotEqual(SUCCESS, view_result)
```
The difference between `assertEqual(SUCCESS` and `assertNotEqual(SUCCESS` is easy to miss, as I think is evidenced by the mistake fixed in https://github.com/dimagi/commcare-hq/pull/28201/commits/2bf07a82f8853d65b2f94420b8b238f7a1e4970e.

Reply:
good call 6f9c7993b321c96c67627387658e83df0e695f5f